geps/gep-2649/index.md

```diff
@@ -0,0 +1,714 @@
+cluster scoped resource, this means that any policy attached to it must also be
+cluster scoped.
+
+GatewayClass parameters provide an alternative to policy attachment that may be
+easier for some implementations to support. These parameters can similarly be
+used to set defaults and requirements for an entire GatewayClass.
+
+### Merging into existing `spec` fields
+
+It's possible (even likely) that configuration in a Policy may need to be merged
+into an existing object's fields somehow, particularly for Inherited policies.
+
+When merging into an existing fields inside an object, Policy objects should
+merge values at a scalar level, not at a struct or object level.
+
+For example, in the `CDNCachingPolicy` example above, the `cdn` struct contains
+a `cachePolicy` struct that contains fields. If an implementation was merging
+this configuration into an existing object that contained the same fields, it
+should merge the fields at a scalar level, with the `includeHost`,
+`includeProtocol`, and `includeQueryString` values being defaulted if they were
+not specified in the object being controlled. Similarly, for `overrides`, the
+values of the innermost scalar fields should overwrite the scalar fields in the
+affected object.
+
+Implementations should not copy any structs from the Policy object directly into the
+affected object, any fields that _are_ overridden should be overridden on a per-field
+basis.
+
+In the case that the field in the Policy affects a struct that is a member of a list,
+each existing item in the list in the affected object should have each of its
+fields compared to the corresponding fields in the Policy.
+
+For non-scalar field _values_, like a list of strings, or a `map[string]string`
+value, the _entire value_ must be overwritten by the value from the Policy. No
+merging should take place. This mainly applies to `overrides`, since for
+`defaults`, there should be no value present in a field on the final object.
+
+This table shows how this works for various types:
+
+|Type|Object config|Override Policy config|Result|
+|----|-------------|----------------------|------|
+|string| `key: "foo"` | `key: "bar"`  | `key: "bar"` |
+|list| `key: ["a","b"]` | `key: ["c","d"]` | `key: ["c","d"]` |
+|`map[string]string`| `key: {"foo": "a", "bar": "b"}` | `key: {"foo": "c", "bar": "d"}` | `key: {"foo": "c", "bar": "d"}` |
+
+### Conflict Resolution
+
+Other conflict resolution between Policy objects must also follow the rules
+outlined in GEP-713, in its Conflict Resolution section.
+
+## Policy Attachment examples and behavior
+
+This approach is building on concepts from all of the alternatives discussed
+below. This is very similar to the (now removed) BackendPolicy resource in the API,
+but also borrows some concepts from the [ServicePolicy
+proposal](https://github.com/kubernetes-sigs/gateway-api/issues/611).
+
+### Policy Attachment for Ingress
+Attaching a Directly Attached Policy to Gateway resources for ingress use cases
```

Comment:
Should this example be moved to the GEP on direct policy attachment?

Reply:
Interestingly, under the criteria as I've updated them, this is no longer a Direct Attached Policy, because it affects Routes and attaches to a Gateway. I'll update the wording.